lua/telescope/builtin/__lsp.lua

```diff
@@ -359,7 +359,11 @@ lsp.document_symbols = function(opts)
       return
     end
 
-    local locations = vim.lsp.util.symbols_to_items(result or {}, opts.bufnr) or {}
+    if vim.fn.has "nvim-0.11" == 1 then
+      local locations = vim.lsp.util.symbols_to_items(result or {}, opts.bufnr, "utf-8") or {}
```

Comment:
I don't think you can hard code the encoding value as different clients may use different encoding. You should get the encoding from the `vim.lsp.Client` object. The request handler function here is passed a third parameter `ctx` (https://neovim.io/doc/user/lsp.html#lsp-handler) which contains the `ctx.client_id`. This can be used to get the client object (`vim.lsp.get_client_by_id(ctx.client_id)`) and access the encoding as `client.offset_encoding`.